src/main/java/sigma/Sigma.java

```diff
@@ -0,0 +1,422 @@
+    private String handleEventCommand(String userInput) throws SigmaException {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yy HH:mm");
+            String[] parts = userInput.split(" /from ");
+            Task event = createEvent(parts, formatter);
+            taskList.addTask(event);
+            return this.ui.showAddedTask(event, taskList.getSize());
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new SigmaInvalidArgException(CommandType.EVENT);
+        } catch (DateTimeParseException e) {
+            throw new SigmaInvalidDateException(CommandType.EVENT);
+        }
+    }
+
+    /**
+     * Creates an event task from the provided input parts and date-time formatter.
+     *
+     * @param parts An array containing the description and time range of the event,
+     *              split by the "/from" and "/to" delimiters.
+     * @param formatter The formatter used to parse the date-time strings.
+     * @return The created event task.
+     * @throws SigmaMissingArgException If any required argument (description, start time, or end time) is missing.
+     * @throws SigmaInvalidDateRangeException If the end time is before the start time.
+     */
+    private static Task createEvent(String[] parts, DateTimeFormatter formatter) throws SigmaMissingArgException,
+            SigmaInvalidDateRangeException {
+        if (parts.length < 2) {
+            throw new SigmaMissingArgException(CommandType.EVENT);
+        }
+        String description = parts[0].trim();
+        String[] timeParts = parts[1].split(" /to ");
+        if (timeParts.length < 2) {
+            throw new SigmaMissingArgException(CommandType.EVENT);
+        }
+        LocalDateTime from = LocalDateTime.parse(timeParts[0], formatter);
+        LocalDateTime to = LocalDateTime.parse(timeParts[1], formatter);
+        if (description.isEmpty()) {
+            throw new SigmaMissingArgException(CommandType.EVENT);
+        }
+        if (to.isBefore(from)) {
+            throw new SigmaInvalidDateRangeException();
+        }
+        return new Event(description, from, to);
+    }
+
+    /**
+     * Handles the "delete" command, removing a task from the list.
+     *
+     * @param userInput The task number to delete.
+     * @throws SigmaException If the task number is invalid.
+     */
+    private String handleDeleteCommand(String userInput) throws SigmaException {
+        try {
+            int taskNumber = Integer.parseInt(userInput);
+            if (taskNumber < 1 || taskNumber > taskList.getSize() || taskList.getSize() == 0) {
+                throw new SigmaInvalidTaskException(taskNumber);
+            }
+            assert taskList.getSize() != 0;
+            String s = this.ui.showDeletedTask(taskList.getTask(taskNumber), taskList.getSize() - 1);
+            taskList.deleteTask(taskNumber);
+            assert taskList.getSize() == (taskList.getSize() + 1) - 1;
+            return s;
```

Comment:
I suggest splitting some of the logic here by inserting newlines between them so that beginners like me can read it easier.